### PR TITLE
aplx6 Mediapad Custom Designed PCB

### PR DESCRIPTION
**Pull request by Aplyard**
A stylish (2x3) 6-key MediaPad for your music and browser shortcuts.

## Description

I designed a PCB and i want to flash it with qmk. i forked your repository, worked on a similar 6key pad to fit my needs. Now i would like my layout to be added to the master branch.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,6 @@
 *.bin
 *.eep
 *.elf
-*.hex
```
Comment (zvecr) [inline .gitignore:7]: This change should be reverted

File: keyboards/aplx6/Hex Test Compilation/aplx6_Media.hex [hunk 2]
```diff
@@ -0,0 +1,1028 @@
+:100000000C941C010C9463010C9463010C94630127
```
Comment (zvecr) [inline keyboards/aplx6/Hex Test Compilation/aplx6_Media.hex:1]: This file should be deleted

File: keyboards/aplx6/config.h [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
+#define CONFIG_H
+
```
Comment (zvecr) [inline keyboards/aplx6/config.h:20]:
```suggestion
#pragma once

```

File: keyboards/aplx6/config.h [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+//#define RGBLIGHT_ANIMATIONS
+//#define RGBLED_NUM X    // Number of LEDs
+
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+#define TAPPING_TERM 200
+
+#endif
```
Comment (zvecr) [inline keyboards/aplx6/config.h:53]:
```suggestion
```

File: keyboards/aplx6/aplx6.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
+//#ifndef APLX6_H
+//#define APLX6_H
+
+#pragma once
```
Comment (zvecr) [inline keyboards/aplx6/aplx6.h:4]:
```suggestion
#pragma once
```

File: keyboards/aplx6/keymaps/Media/keymap.c [hunk 6]
```diff
@@ -0,0 +1,38 @@
+#include QMK_KEYBOARD_H
```
Comment (zvecr) [inline keyboards/aplx6/keymaps/Media/keymap.c:1]: This keymap should either be renamed to default, or renamed to lower case and a default keymap added

File: keyboards/aplx6/README.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+# Aplx6
+https://github.com/Aplyard/Aplx6
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:2]: Would like this file to be more aligned to the QMK template.

```suggestion

![Front](https://i.imgur.com/lUg9GAo.png)  
![Back](https://i.imgur.com/iAHtHV7.png)
``````

File: keyboards/aplx6/README.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
+# Aplx6
+https://github.com/Aplyard/Aplx6   
+   
+A stylish (2x3) 6-key MediaPad for your music and browser shortcuts.
+Designed to be assembled as a sandwich with a blank pcb, using m2.5 screws and spacers. 
+Diodes : 1N4148
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:6]: Some rewording.

```suggestion
A stylish (2x3) 6-key MediaPad for your music and browser shortcuts. Designed to be assembled as a sandwich with a blank PCB, using M2.5 screws and spacers, your choice of MX- or Alps-compatible switches, and 1N4148 diodes.
```

File: keyboards/aplx6/README.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+---
+
+![Front](https://i.imgur.com/lUg9GAo.png)
+
+![Back](https://i.imgur.com/iAHtHV7.png)
+
+![Blank](https://i.imgur.com/EnqIhSW.png)
+
+---
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:16]:
```suggestion
```

File: keyboards/aplx6/README.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+**Can support Rgb underglow (WS2812)**
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:18]:
```suggestion
## RGB Underglow (WS2812)
```

File: keyboards/aplx6/README.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+---
+
+**Can support Rgb underglow (WS2812)** 
+
+Just wire them to any of the unused ProMicro pins.
+Don't forget to edit-uncomment the [config.h](https://github.com/Aplyard/Aplx6/blob/master/qmk/config.h) and the [rules.mk](https://github.com/Aplyard/Aplx6/blob/master/qmk/rules.mk)
+for RGB underglow support or even add your own functions and modes.
+Used pins can be found in [config.h](https://github.com/Aplyard/Aplx6/blob/master/qmk/config.h) or in the [KiCad Schematic](https://github.com/Aplyard/Aplx6/blob/master/kiCad/xd6.sch)
+All ProMicro pins can be found [here](https://golem.hu/article/pro-micro-pinout/) along with the +2 pins mod.
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:24]:
```suggestion
Just wire them to any of the unused ProMicro pins. Don't forget to edit-uncomment the `config.h` and the `rules.mk` for RGB underglow support, or even add your own functions and modes. Used pins can be found in `config.h` or in the [KiCad Schematic](https://github.com/Aplyard/Aplx6/blob/master/kiCad/xd6.sch). All ProMicro pins can be found [here](https://golem.hu/article/pro-micro-pinout/) along with the +2 pins mod.
```

File: keyboards/aplx6/README.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+---
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:26]:
```suggestion
* Keyboard Maintainer: Aplyard
* Hardware Supported: Aplx6 PCB, Pro Micro (ATmega32U4)
* Hardware Availability: Aplyard
```

File: keyboards/aplx6/README.md [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+|Prev|VolDown|Next|
+
+=========================
+| Fn |Layer|(1)
+|--|--|--|
+|Calc|PgUp|KC_TRANS|
+|MyPc|PgDown|RESET
+
+---
+
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:44]: This should be in `keyboards/aplx6/keymaps/default/readme.md`.

```suggestion
```

File: keyboards/aplx6/README.md [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+Make example for this keyboard (after setting up your build environment):
+
+     make aplx6:Media
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:47]:
```suggestion
     make aplx6:default
```

File: keyboards/aplx6/README.md [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+
+See [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) then the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information.
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:49]:
```suggestion
See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/aplx6/README.md [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+
+**Alternatives of qmk that worked for me:**  
+[Keyboard Layout Editor](http://www.keyboard-layout-editor.com/#/)   
+[Keyboard Plate & Case Builder](http://builder.swillkb.com/)  
+[Firmware Builder](https://kbfirmware.com/)      
+
+---   
+Keyboard Maintainer: Aplyard  
+Hardware Supported: Pro Micro ATmega32U4  
+Hardware Availability: Aplyard
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:72]:
```suggestion
```

File: keyboards/aplx6/rules.mk [hunk 17]
```diff
@@ -0,0 +1,32 @@
...
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in 
+#   the appropriate keymap folder that will get included automatically
```
Comment (noroadsleft) [inline keyboards/aplx6/rules.mk:16]:
```suggestion
#   change yes to no to disable
```

File: keyboards/aplx6/rules.mk [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no       # Enable WS2812 RGB underlight.
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (noroadsleft) [inline keyboards/aplx6/rules.mk:32]: Aligns this section to the [QMK AVR template for this file](https://github.com/qmk/qmk_firmware/blob/be2f5816b6cf6513591f825df1d0438729b83388/quantum/template/avr/rules.mk).

```suggestion
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
UNICODE_ENABLE = yes        # Unicode
```

**Comment by Aplyard** [comment]
Last updates(51cee7f & 94e34cf) are 2x Rendered photos and 2 pins changed due to fabrication reasons. Sorry for the inconvenience.

File: keyboards/aplx6/README.md [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+![Back](https://i.imgur.com/PXqNmUh.png)
+   
+A stylish (2x3) 6-key MediaPad for your music and browser shortcuts. Designed to be assembled as a sandwich with a blank PCB, using M2.5 screws and spacers, your choice of MX- or Alps-compatible switches, and 1N4148 diodes.
+
+
+## RGB Underglow (WS2812)
+
+Just wire them to any of the unused ProMicro pins. Don't forget to edit-uncomment the `config.h` and the `rules.mk` for RGB underglow support, or even add your own functions and modes. Used pins can be found in `config.h` or in the [KiCad Schematic](https://github.com/Aplyard/Aplx6/blob/master/kiCad/xd6.sch). All ProMicro pins can be found [here](https://golem.hu/article/pro-micro-pinout/) along with the +2 pins mod.
+
+* Keyboard Maintainer: Aplyard
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:13]:
```suggestion
* Keyboard Maintainer: [Aplyard](https://github.com/Aplyard)
```

File: keyboards/aplx6/README.md [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+* Hardware Supported: Aplx6 PCB, Pro Micro (ATmega32U4)
+* Hardware Availability: Aplyard
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:15]: Meant to paste this here before; my fault.

```suggestion
* Hardware Availability: [GitHub](https://github.com/Aplyard/Aplx6)
```

File: keyboards/aplx6/README.md [hunk 21]
```diff
@@ -0,0 +1,41 @@
...
+
+**Usefull Links:**  
+[Qmk Online Configurator](https://config.qmk.fm/#)  
+
+**Alternatives of qmk that worked for me:**  
+[Keyboard Layout Editor](http://www.keyboard-layout-editor.com/#/)   
+[Keyboard Plate & Case Builder](http://builder.swillkb.com/)  
+[Firmware Builder](https://kbfirmware.com/)      
+
+---
```
Comment (noroadsleft) [inline keyboards/aplx6/README.md:41]: Cleans up the markdown, and some spelling corrections.

```suggestion
**Designed in:**  
[KiCad](https://github.com/KiCad)  

**Components Footprints & Libraries:**  
[keebs.pretty](https://github.com/egladman/keebs.pretty)  
[keyboard_parts.pretty](https://github.com/tmk/keyboard_parts.pretty)  
[ProMicro KiCad](https://github.com/Biacco42/ProMicroKiCad)  

**Useful Links:**  
[QMK Online Configurator](https://config.qmk.fm/#)  

**Alternatives of QMK that worked for me:**  
[Keyboard Layout Editor](http://www.keyboard-layout-editor.com/#/)  
[Keyboard Plate & Case Builder](http://builder.swillkb.com/)  
[Firmware Builder](https://kbfirmware.com/)  
```

**Comment by Aplyard** [comment]
all changes are made, thanks for your suggestions and your help. It's up to you know :)

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**